api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -335,50 +396,48 @@ def create_wgs_extraction_dataset(cmd_name, args)
     "--ttl [ttl]",
     ->(opts, v) { opts.ttl = v},
     "Add default ttl to dataset tables. Given in seconds.")
-  op.add_validator ->(opts) { raise ArgumentError unless opts.bq_dataset and opts.project and opts.tier and opts.ttl }
+  op.add_validator ->(opts) { raise ArgumentError unless opts.project and opts.tier and opts.ttl }
   op.add_validator ->(opts) { raise ArgumentError.new("unsupported project: #{opts.project}") unless ENVIRONMENTS.key? opts.project }
   op.add_validator ->(opts) { raise ArgumentError.new("unsupported tier: #{opts.tier}") unless ENVIRONMENTS[opts.project][:accessTiers].key? opts.tier }
   op.parse.validate
 
   common = Common.new
   env = ENVIRONMENTS[op.opts.project]
-  account = env.fetch(:publisher_account)
-  tier = env.fetch(:accessTiers)[op.opts.tier]
-
-  # TODO(RW-3208): Investigate using a temporary / impersonated SA credential instead of a key.
-  key_file = Tempfile.new(["#{account}-key", ".json"], "/tmp")
-  ServiceAccountContext.new(
-    op.opts.project, account, key_file.path).run do
-    # TODO(RW-3768): This currently leaves the user session with an activated service
-    # account user. Ideally the activation would be hermetic within the docker
-    # session, or else we would revert the active account after running.
-    common.run_inline %W{gcloud auth activate-service-account -q --key-file #{key_file.path}}
-
-    dest_dataset = "#{tier.fetch(:dest_cdr_project)}:#{op.opts.bq_dataset}"
-    common.run_inline %W{bq mk --default_table_expiration #{op.opts.ttl}  --dataset #{dest_dataset}}
+  proxy_group = must_get_wgs_proxy_group(op.opts.project)
+  cdr_project = env.fetch(:accessTiers)[op.opts.tier].fetch(:dest_cdr_project)
+
+  extract_config = get_config(op.opts.project).fetch("wgsCohortExtraction", {})
+  fq_datasets = [
+    "extractionCohortsDataset",
+    "extractionDestinationDataset",
+    "extractionTempTablesDataset"
+  ].map do |key|
+    fq_ds = extract_config.fetch(key)
+    raise ArgumentError.new("missing config value for #{key} in project #{op.opts.project}") unless fq_ds
+    raise ArgumentError.new("config value (#{fq_ds}) doesn't match expected CDR project for tier (#{cdr_project})") unless fq_ds.start_with? "#{cdr_project}."
+    fq_ds.sub("\.", ":")
+  end
```

Comment:
@ericsong PTAL, I started creating the tmp dataset and found it too tedious. Instead - I've refactored this command to instead just create/update all WGS datasets for a given environment, as specified in the config. So the steps would be:

1. add dataset names to the config
2. run `create-wgs-extraction-datasets`

I've verified this works in staging.